sgx/src/attestation_types/mod.rs

```diff
@@ -4,9 +4,7 @@
 //! Section references in further documentation refer to this document.
 //! https://www.intel.com/content/dam/www/public/emea/xe/en/documents/manuals/64-ia-32-architectures-software-developer-vol-3d-part-4-manual.pdf
 
-#![deny(clippy::all)]
-#![allow(clippy::identity_op)]
-#![deny(missing_docs)]
+#![cfg(feature = "std")]
```

Comment:
Not sure if this changes in one of the later commits, but since the `attestation-types` module will be used inside the enclave, it should not be `std`.

Reply:
Hmm, that's interesting because currently `attestation-types` requires `std`. Anyhow, I'm fine in adapting it to `core`, but perhaps we should make it in a subsequent PR?

Reply:
Unless, the changes are trivial but I'll need to take a closer look :-)

Reply:
@kubkon That *is* interesting! In that case, feel free to leave as-is and we can file a separate issue for it when it comes up.

Reply:
@lkatalin OK, I've had a closer look, and it seems we're using `std::vec::Vec` which means, for the module to be `no_std` compatible we'd either need to replace any structure requiring a global allocator with a statically-sized equivalent such as a slice, or write our own alloc, and then we are OK to use `alloc::vec::Vec`. Note however that in the latter case we're forced to rely on unstable nightly features since not everything has been stabilised in Rust wrt to custom global allocators (in fact, I'm thinking here mainly of [this issue](https://github.com/rust-lang/rust/issues/51540)). FWIW, I believe that we could easily write our own "fixed memory allocator" which would essentially be a wrapper around a fixed chunk of statically-sized memory we could operate on. Either way, I think this warrants a separate issue and a broader discussion, would you agree?

Reply:
FYI @lkatalin, I've moved the discussion to #755.

Reply:
That's great, thanks for filing the issue @kubkon .